File: python/novax_sdk/pay/h5/h5_confirm_request.py
from typing import Any, Optional

from novax_sdk.request.api_request import ApiRequest, HttpMethod


class H5ConfirmRequest(ApiRequest):
    """POST /pay/public/h5/confirm"""

    def __init__(self, protocol: Optional[str] = None, currency: Optional[str] = None,
                 smart_contract_address: Optional[str] = None,
                 company_user_id: Optional[str] = None,
                 token: Optional[str] = None) -> None:
        self._protocol = protocol
        self._currency = currency
        self._smart_contract_address = smart_contract_address
        self._company_user_id = company_user_id
        self._token = token

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.POST

    @property
    def path(self) -> str:
        return "/pay/public/h5/confirm"

    def body(self) -> dict[str, Any]:
        return {
            "protocol": self._protocol,
            "currency": self._currency,
            "smartContractAddress": self._smart_contract_address,
            "companyUserId": self._company_user_id,
            "token": self._token,
        }

    @property
    def response_type(self):
        return None

File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/h5/H5ConfirmRequest.java
package com.novax.sdk.pay.h5;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * POST /pay/public/h5/confirm — user clicks "confirm" on the H5 pay page.
 */
@Getter
@Builder
public class H5ConfirmRequest extends AbstractApiRequest<Object> {

    private final String protocol;
    private final String currency;
    private final String smartContractAddress;
    private final String companyUserId;
    private final String token;

    @Override
    public HttpMethod method() {
        return HttpMethod.POST;
    }

    @Override
    public String path() {
        return "/pay/public/h5/confirm";
    }

    @Override
    public Object body() {
        Map<String, Object> b = new LinkedHashMap<>();
        b.put("protocol", protocol);
        b.put("currency", currency);
        b.put("smartContractAddress", smartContractAddress);
        b.put("companyUserId", companyUserId);
        b.put("token", token);
        return b;
    }

    @Override
    public TypeRef<Object> responseType() {
        return TypeRef.of(Object.class);
    }
}
